Runtime/Scripts/DeviceInfoIntegration.cs

```diff
@@ -0,0 +1,150 @@
+      {
+        Debug.Log("No TelephonyManager!");
+        return null;
+      }
+      map = new Dictionary<string, string>();
+
+      AndroidJavaClass versionClass = new AndroidJavaClass("android.os.Build$VERSION");
+      int sdk_int = PlatformIntegrationUtil.GetStatic<int>(versionClass, "SDK_INT");
+      map["Build.VERSION.SDK_INT"] = sdk_int.ToString();
+
+      const string readPhoneStatePermissionString = "READ_PHONE_STATE";
+      try
+      {
+        if (!Permission.HasUserAuthorizedPermission(readPhoneStatePermissionString))
+        {
+          string ver = PlatformIntegrationUtil.Call<string>(telephonyManager, "getDeviceSoftwareVersion");
+          if (ver != null)
+          {
+            map["DeviceSoftwareVersion"] = ver.ToString();
+          }
+        }
+      }
+      catch (Exception e)
+      {
+        Debug.Log("Exception retrieving properties: " + e.GetBaseException() + ", " + e.Message);
+      }
+
+      try
+      {
+        if (!Permission.HasUserAuthorizedPermission(readPhoneStatePermissionString))
+        {
+          int nType = PlatformIntegrationUtil.Call<int>(telephonyManager, "getDataNetworkType");
+          NetworkDataType datatype = (NetworkDataType)nType;
+          map["DataNetworkType"] = datatype.ToString();
+        }
+      }
+      catch (Exception e)
+      {
+        Debug.Log("Exception retrieving properties: " + e.GetBaseException() + ", " + e.Message);
+      }
+
+      AndroidJavaClass versionCodesClass = new AndroidJavaClass("android.os.Build$VERSION_CODES");
+      int versionCode = PlatformIntegrationUtil.GetStatic<int>(versionCodesClass, "Q");
+      if (sdk_int > versionCode)
+      {
+        string mc = PlatformIntegrationUtil.Call<string>(telephonyManager, "getManufacturerCode");
+        if (mc != null)
+        {
+          map["ManufacturerCode"] = mc;
+        }
+      }
+
+      string niso = PlatformIntegrationUtil.Call<string>(telephonyManager, "getNetworkCountryIso");
+      if (niso != null)
+      {
+        map["NetworkCountryIso"] = niso;
+      }
+
+      string siso = PlatformIntegrationUtil.Call<string>(telephonyManager, "getSimCountryIso");
+      if (siso != null)
+      {
+        map["SimCountryCodeIso"] = siso;
+      }
+
+      int phoneType = PlatformIntegrationUtil.Call<int>(telephonyManager, "getPhoneType");
+      map["PhoneType"] = phoneType.ToString();
+
+      // Default one.
+      string simOperatorName = PlatformIntegrationUtil.Call<string>(telephonyManager, "getSimOperatorName");
+      if (simOperatorName != null)
+      {
+        map["SimOperator"] = simOperatorName;
+      }
+
+      // Default one.
+      string networkOperator = PlatformIntegrationUtil.Call<string>(telephonyManager, "getNetworkOperatorName");
+      if (networkOperator != null)
+      {
+        map["NetworkOperatorName"] = networkOperator;
+      }
+
+      return map;
+    }
+#elif UNITY_IOS
+  Dictionary<string, string> GetDeviceDetails()
```

Comment:
GetDeviceInfo() instead of GetDeviceDetails() as it's a required member by DeviceInfo Interface.

Reply:
Also it needs to be a public function

Reply:
done.